filters/auth/oidc.go

```diff
@@ -763,6 +763,15 @@ func (f *tokenOidcFilter) tokenClaims(ctx filters.FilterContext, oauth2Token *oa
 		return nil, "", requestErrorf("claims do not contain sub")
 	}
 
+	exp, ok := tokenMap["exp"].(float64)
+	if ok {
+		val := time.Until(time.Unix(int64(exp), 0))
+		if val > time.Minute {
+			f.validity = val
+			log.Debugf("Setting validity of oidc filter to %s", f.validity)
+		}
+	}
+
```

Comment:
This does not seem as a correct approach. The filter instance serves multiple concurrent requests so it should be immutable in this case. 

The better approach seems to be to pass validity/max age `max age = claims[exp] or f.validity` to `doDownstreamRedirect` where cookies are created.

Reply:
Another thing to verify is whether `claims["exp"]` is always unmarshaled as `float64` otherwise we may need a bit more sophisticated time conversion logic.

Reply:
> is always unmarshaled as float64

I think existing test is enough